src/app/vault/add-edit.component.ts

```diff
@@ -169,4 +175,14 @@ export class AddEditComponent extends BaseAddEditComponent {
             await this.totpUpdateCode();
         }
     }
+
+    private async getCipherCollectionNames(): Promise<string[]> {
+        const collectionNames: string[] = [];
+        for (const id of this.cipher.collectionIds) {
+            const collection = await this.collectionService.get(id);
+            const name = await collection.name.decrypt(this.cipher.organizationId)
+            collectionNames.push(name)
+        }
+        return collectionNames.sort()
+    }
```

Comment:
This should probably be put into a service method in the collectionService. Maybe a service method for "get many by cipher id".

Reply:
If I move this into a service method I have to edit the jslib submodule project, right? Do I have to create a new branch there and create a separated pull request in the jslib project? Since I do not have any experience in working with git submodules I would be really grateful if someone could give me some advice here.

Reply:
Hi @mauamy, not sure if anyone has gotten back to you on this, however yes, moving this up to the service method would mean updating jslib. Essentially when you pull this repo down, the git submodule is initialized (to latest) after running `npm install`, you can simply edit-in place, and you can fork/branch right from within the submodule directory pointing it to a new upstream, etc. That way you can test w/o too much complication. You would run your git commands and push to your fork from your local repo of jslib and submit that PR separately, then link it to the related PR from any changes needed in web. If it's a breaking change in the jslib service interface/abstract method, etc. things get hairy, but otherwise adding a new method hopefully shouldn't cause too much heartburn.

Reply:
@cscharf Thank you very much for your help!
I now created the recommended method in the collection service and created a separate pull request like you proposed.